src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.example.bookwebstoryspringboot.repository.impl;
+
+import com.example.bookwebstoryspringboot.entity.Book;
+import com.example.bookwebstoryspringboot.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.TypedQuery;
+import java.util.List;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    private EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction entityTransaction = null;
+
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            entityTransaction = entityManager.getTransaction();
+            entityManager.persist(book);
+            entityTransaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (entityTransaction != null) {
+                entityTransaction.rollback();
+            } else {
+                throw new RuntimeException("Can't save " + book + " " + e);
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            TypedQuery<Book> query = entityManager.createQuery("SELECT b.id, b.description,"
+                    + "b.title,b.author,"
+                    + "b.coverImage, b.isbn, b.price FROM Book b", Book.class);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("there are no books");
+
+        }
+
```

Comment:
remove redundant empty line
```suggestion
        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
            TypedQuery<Book> query = entityManager.createQuery("SELECT b.id, b.description,"
                    + "b.title,b.author,"
                    + "b.coverImage, b.isbn, b.price FROM Book b", Book.class);
            return query.getResultList();
        } catch (Exception e) {
            throw new RuntimeException("there are no books");
        }
```